ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -154,24 +154,22 @@ class MySqlNamedBlobDb implements NamedBlobDb {
    * It will pull out any stale record (limit to be config.queryStaleDataMaxResults [Default 1000] records at most)
    * meeting below conditions:
    * 1. created more than config.staleDataRetentionDays [Default 5] days ago, and
-   * 2. it's NOT a VLR (Valid Latest Record) for any named blob.
+   * 2. its blob_id does NOT show in any VLR (Valid Latest Record) of any named blob.
    * VLR of a named blob is the record whose blob_state=1 (READY), and has the max version for the named blob.
    *
-   * To pull out the rows that meet the 2nd condition above, we are using left join and then pull out the rows where
-   * right side table's keys are all NULL
    */
   // @formatter:off
   private static final String GET_STALE_QUERY = String.format(""
-          + "SELECT t1.%s, t1.%s, t1.%s, t1.%s, t1.%s, t1.%s, %s "
-          + "FROM %s t1 "
-          + "LEFT JOIN "
-          + "(SELECT account_id, container_id, blob_name, max(version) as version "
+          + "SELECT %s, %s, %s, %s, %s, %s, %s "
```

Comment:
It's more complicated because we need to prevent deleting valid blob_id. The previous SQL doesn't prevent that. 
Since blob_id are indexed, it runs pretty fast:
```
mysql> SELECT account_id, container_id, blob_name, blob_id, version, deleted_ts, CURRENT_TIMESTAMP(6) FROM named_blobs_v2 WHERE blob_id not in (SELECT distinct(blob_id)   FROM named_blobs_v2   WHERE version in   (SELECT max(version) as version   FROM named_blobs_v2   WHERE blob_state = 1 and version != 0   GROUP BY account_id, container_id, blob_name) ) AND version<168680160000000000 ORDER BY blob_id LIMIT 10000;
```
It's completed ~6 seconds:

```
CC3FE4E7DD38CD9 |                  0 | NULL                       | 2023-06-23 18:17:19.738068 |
+------------+--------------+--------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------+--------------------+----------------------------+----------------------------+
10000 rows in set (6.11 sec)
```